btf/ext_info.go

```diff
@@ -142,12 +142,15 @@ func AssignMetadataToInstructions(
 
 // MarshalExtInfos encodes function and line info embedded in insns into kernel
 // wire format.
+//
+// If an instruction has an [asm.Comment], it will be synthesized into a mostly
+// empty line info.
 func MarshalExtInfos(insns asm.Instructions, b *Builder) (funcInfos, lineInfos []byte, _ error) {
 	iter := insns.Iterate()
 	for iter.Next() {
-		_, ok := iter.Ins.Source().(*Line)
+		source := iter.Ins.Source()
 		fn := FuncMetadata(iter.Ins)
-		if ok || fn != nil {
```

Comment:
I could simplify the check here, because `.Source()` returns a `fmt.Stringer` and the code below now works with any non-nil `fmt.Stringer`, not just `*Line`s.